### PR TITLE
pow2: header file for pow2 solver

### PR DESCRIPTION
**Pull request by yoni206**
This PR adds a header file for the pow2 solver. It also includes an empty test file, to trigger compilation of the header file. The next PR will include implementations and tests.

### CODE REVIEW

File: src/theory/arith/nl/pow2_solver.h [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+namespace nl {
+
+class NlModel;
+
+/** pow2 solver class
+ *
+ */
+class Pow2Solver
+{
+  typedef context::CDHashSet<Node> NodeSet;
```
Comment (ajreynol) [inline src/theory/arith/nl/pow2_solver.h:40]:
```suggestion
  using NodeSet = context::CDHashSet<Node>;
```

File: src/theory/arith/nl/pow2_solver.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+   */
+  void checkFullRefine();
+
+  //-------------------------------------------- end lemma schemas
+ private:
+  // The inference manager that we push conflicts and lemmas to.
+  InferenceManager& d_im;
+  /** Reference to the non-linear model object */
+  NlModel& d_model;
+  /** commonly used terms */
```
Comment (ajreynol) [inline src/theory/arith/nl/pow2_solver.h:83]: Do you need all of these?

File: src/theory/arith/nl/pow2_solver.h [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  Node d_false;
+  Node d_true;
+  Node d_zero;
+  Node d_one;
+  Node d_two;
+
+  NodeSet d_initRefine;
+  /** all pow2 terms */
```
Comment (ajreynol) [inline src/theory/arith/nl/pow2_solver.h:91]: Give the scope of these: I guess this list is to be cleared/populated during full effort checks?  Or should this be context-dependent?

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/theory/arith/nl/pow2_solver.h (hunk 2 above)
Comment (yoni206) [reply src/theory/arith/nl/pow2_solver.h:83]: I will in the next PR:
https://github.com/yoni206/cvc5/blob/15cf5bb36d88fa153be39d7738a47de2412af187/src/theory/arith/nl/pow2_solver.cpp#L94
https://github.com/yoni206/cvc5/blob/15cf5bb36d88fa153be39d7738a47de2412af187/src/theory/arith/nl/pow2_solver.cpp#L105

File: src/theory/arith/nl/pow2_solver.h (hunk 3 above)
Comment (ajreynol) [reply src/theory/arith/nl/pow2_solver.h:91]: Do document how these are updated.

**Review by ajreynol — CHANGES_REQUESTED** [review]

**Comment by yoni206** [comment]
Oops, added it to the wrong branch, sorry :)
now done in this branch, here:  2e28f97

**Review by ajreynol — APPROVED** [review]

**Pull request merged**